TeamCode/src/main/java/com/acmerobotics/relicrecovery/motion/MotionState.java

```diff
@@ -0,0 +1,166 @@
+package com.acmerobotics.relicrecovery.motion;
+
+/**
+ *  @author kellyrm
+ *  fully defined state of one dimensional motion
+ */
+
+public class MotionState {
+
+    /**
+     * position
+     */
+    public double x;
+
+    /**
+     * velocity - x/t
+     */
+    public double v;
+
+    /**
+     * acceleration - x/t^2
+     */
+    public double a;
+
+    /**
+     * jerk - x/t^3
+     */
+    public double j;
+
+    /**
+     * time
+     */
+    public double t;
+
+
+    public MotionState (double x, double v, double a, double j, double t) {
+        this.x = x;
+        this.v = v;
+        this.a = a;
+        this.j = j;
+        this.t = t;
+    }
+
+    /**
+     * flip the motion state
+     * @return a new motion state with everything negated
+     */
+    public MotionState flipped() {
+        return new MotionState(-x, -v, -a, -j, t);
+    }
+
+
+    /**
+     * get all times that the current state will reach a position
+     * @param pos the position to check for
+     * @return the times it will be at pos - returns {} if it will never be at pos or if it is always at pos
+     */
+    public double[] timesAtPos(double pos) {
+        //third degree case
+        if (j != 0) {
+            //according to the internet this is a way to find roots of cubic functions, and it appears to work
+            //don't ask be what is going on
+            //i'm not quite sure why there are trig functions involved
```

Comment:
For reference, here's the formula (w/ a derivation): https://en.wikipedia.org/wiki/Cubic_function#Trigonometric_solution_for_three_real_roots